store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -6147,6 +6147,11 @@ private void checkPasswordStrength(String password, Account acct, Cos cos, ZMuta
                     new Argument(Provisioning.A_zimbraPasswordMaxLength, maxLength, Argument.Type.NUM));
         }
 
+        if (LC.password_contains_username.booleanValue()
+                && StringUtils.containsIgnoreCase(password, acct.getUCUsername())) {
+            throw AccountServiceException.INVALID_PASSWORD("password contains username");
```

Comment:
on UI side, are we just going to handle this exception or we will be checking this logic on UI side as well?
recently we added these password related logics to UI screens also so user experience can be improved

Reply:
I think on UI side we only need to handle exceptions, as the check is based on a local config value for which check is happening on the backend side. I am not sure the same check can be implemented on the UI side or not
@dawoodshaikh can better answer the same

Reply:
@silentsakky @dawoodshaikh , I have changed the error code to the existing error code being used in password scenarios and this is the error message being displayed, I think we would not require UI work on this

Could you please confirm same

![image](https://user-images.githubusercontent.com/96303121/160345423-d175fe12-7eeb-4670-b701-9edaedab24dd.png)

Reply:
@silentsakky , As a discussed updated the GetInfo call to include `zimbraPasswordAllowUsername` in the response
Also updated the Exception statement in checkPasswordStrength() for username case to pass on the argument which can be used on UI side `zimbraPasswordAllowUsername`

GetInfo response:
![image](https://user-images.githubusercontent.com/96303121/160461760-4af402ad-9364-40e4-b850-7ee8dd77e24e.png)

Password change response which contains keyword which can be used to show exact error on UI side
`{"Header":{"context":{"session":{"id":"12","_content":"12"},"change":{"token":1499},"_jsns":"urn:zimbra"}},"Body":{"Fault":{"Code":{"Value":"soap:Sender"},"Reason":{"Text":"invalid password: password contains username"},"Detail":{"Error":{"Code":"account.INVALID_PASSWORD","Trace":"qtp510109769-144:1648491353457:ffcdb33d43d3cabf","a":[{"n":"zimbraPasswordAllowUsername","t":"STR","_content":""}],"_jsns":"urn:zimbra"}}}},"_jsns":"urn:zimbraSoap"}`

Reply:
@silentsakky , updated the Getinfo response `zimbraPasswordAllowUsername` to return in capaitalize form